e2e-test/cypress/integration/workflowitem_permissions_spec.js

```diff
@@ -1,82 +1,420 @@
+    // Remove permission
+    changePermissionInGui("workflowitem.intent.grantPermission", testUser.id);
+    cy.get("[data-test=permission-submit]").click();
+    cy.get("[data-test=confirmation-dialog-cancel]").should("not.be.visible");
+
+    // Reset permissions
+    cy.login("root", "root-secret");
+    cy.grantWorkflowitemPermission(projectId, subprojectId, workflowitemId, intent, testUser.id);
+    cy.grantWorkflowitemPermission(projectId, subprojectId, workflowitemId, intent, executingUser.id);
+  });
+
+  it("Submitting the permission dialog without workflowitem.intent.revokePermission disables the submit button when removing user", function() {
+    const grantIntent = "workflowitem.intent.grantPermission";
+    const revokeIntent = "workflowitem.intent.revokePermission";
+
+    cy.grantWorkflowitemPermission(projectId, subprojectId, workflowitemId, grantIntent, testUser.id);
+    cy.revokeWorkflowitemPermission(projectId, subprojectId, workflowitemId, revokeIntent, executingUser.id);
+
+    cy.get("[data-test=show-workflowitem-permissions]")
+      .first()
+      .click();
+    // Remove permission
+    changePermissionInGui("workflowitem.intent.grantPermission", testUser.id);
+    cy.get("[data-test=permission-submit]").should("be.disabled");
+    // Add permission
+    changePermissionInGui("workflowitem.intent.grantPermission", testUser.id);
+    cy.get("[data-test=permission-submit]").click();
+    cy.get("[data-test=confirmation-dialog-cancel]").should("not.be.visible");
+
+    // Reset permissions
+    cy.grantWorkflowitemPermission(projectId, subprojectId, workflowitemId, revokeIntent, executingUser.id);
+    cy.revokeWorkflowitemPermission(projectId, subprojectId, workflowitemId, grantIntent, testUser.id);
+  });
+
+  it("User having 'view permissions'- permission only can view but not grant/revoke permissions", function() {
+    const grantIntent = "workflowitem.intent.grantPermission";
+    const revokeIntent = "workflowitem.intent.revokePermission";
+
+    cy.revokeWorkflowitemPermission(projectId, subprojectId, workflowitemId, grantIntent, executingUser.id);
+    cy.revokeWorkflowitemPermission(projectId, subprojectId, workflowitemId, revokeIntent, executingUser.id);
+
+    cy.get("[data-test=show-workflowitem-permissions]")
+      .first()
+      .click();
+    cy.get(`[data-test='permission-select-workflowitem.view']`).click();
+    cy.get("[data-test=read-only-permissions-text]").should("be.visible");
+    cy.get("[data-test=permission-list]")
+      .find(`li[value*='${testUser.id}']`)
+      .find("input")
+      .should("be.disabled");
+
+    cy.login("root", "root-secret");
+    cy.grantWorkflowitemPermission(projectId, subprojectId, workflowitemId, grantIntent, executingUser.id);
+    cy.grantWorkflowitemPermission(projectId, subprojectId, workflowitemId, revokeIntent, executingUser.id);
+  });
+
+  it("Granting update permissions views 5 additional permissions needed", function() {
```

Comment:
Is it possible that this changes? Should we really hard code the number 5 in there?

Reply:
Since the tests reset all permissions granted or revoked, there should always be shown 5 actions in the table. If the number of actions are not 5 there is an issue so its intentional